thread/common/threadhelpers.cpp

```diff
@@ -70,17 +70,23 @@ omrthread_spinlock_acquire(omrthread_t self, omrthread_monitor_t monitor)
 	uintptr_t spinCount2Init = monitor->spinCount2;
 	uintptr_t spinCount1Init = monitor->spinCount1;
 
-	uintptr_t spinCount3 = spinCount3Init;
-	uintptr_t spinCount2 = spinCount2Init;
-
 #if defined(OMR_THR_SPIN_WAKE_CONTROL)
- 	if (monitor->spinThreads < lib->maxSpinThreads) {
- 		VM_AtomicSupport::add(&monitor->spinThreads, 1);
- 	} else {
- 		goto exit;
- 	}
+	BOOLEAN spinning = TRUE;
+	if (OMRTHREAD_IGNORE_SPIN_THREAD_BOUND != lib->maxSpinThreads) {
+		if (monitor->spinThreads < lib->maxSpinThreads) {
+			VM_AtomicSupport::add(&monitor->spinThreads, 1);
+		} else {
+			spinCount1Init = 1;
+			spinCount2Init = 1;
```

Comment:
Why are you just not setting the affected variables directly here instead of setting the Init values?

Reply:
the Init values are needed for calculation of JLM counts at the end. Init values need to preserve the spin counts at the beginning of the spin loop.